src/impl/interaction/library/PanTouch.ts

```diff
@@ -0,0 +1,165 @@
+     */
+    public constructor(horizontal: boolean, minLength: number) {
+        super();
+        this.touchID = undefined;
+        this.isHorizontal = horizontal;
+        this.minLength = minLength;
+    }
+
+    /**
+     * Checks if the Pan this PanTouch is part of is horizontal enough (or vertical enough) for the interaction to continue.
+     */
+    private hasCorrectOrientation(dataHandler?: PanTouchFSMHandler): boolean {
+        return this.isHorizontal ? dataHandler?.isHorizontal() === true : dataHandler?.isVertical() === true;
+    }
+
+    /**
+     * Checks if the Pan this PanTouch is part of is long enough to be ended successfully.
+     */
+    private hasCorrectLength(dataHandler?: PanTouchFSMHandler): boolean {
+        if (dataHandler !== undefined) {
+            return this.isHorizontal
+                ? (Math.abs(dataHandler.getDiffX(this.touchID)) >= this.minLength) : (Math.abs(dataHandler.getDiffY(this.touchID)) >= this.minLength);
+        }
+        return false;
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    public override buildFSM(dataHandler?: PanTouchFSMHandler): void {
+        if (this.states.length > 1) {
+            return;
+        }
+        super.buildFSM(dataHandler);
+
+        const touched = new StdState(this, "touched");
+        const moved = new StdState(this, "moved");
+        const released = new TerminalState(this, "released");
+        const cancelled = new CancellingState(this, "cancelled");
+
+        this.addState(touched);
+        this.addState(moved);
+        this.addState(released);
+        this.addState(cancelled);
+
+        // The interaction doesn't start as long as the contact point hasn't started moving.
+        this.startingState = moved;
+
+        this.createTransitions(touched, moved, released, cancelled, dataHandler);
+
+        super.buildFSM(dataHandler);
+    }
+
+    /**
+     * Creates the transitions for the FSM.
+     */
+    private createTransitions(touched: StdState, moved: StdState, released: TerminalState, cancelled: CancellingState,
+                              dataHandler?: PanTouchFSMHandler): void {
+        // The condition that checks if the interaction can continue after any new movement.
+        const moveCondition = (event: TouchEvent): boolean => (event.changedTouches[0].identifier === this.touchID) &&
+            this.hasCorrectOrientation(dataHandler);
+
+        const pressure = new TouchPressureTransition(this.initState, touched);
+        pressure.action = (event: TouchEvent): void => {
+            this.touchID = event.changedTouches[0].identifier;
+            dataHandler?.onTouch(event);
+        };
+
+        const initialMoveOK = new TouchMoveTransition(touched, moved);
+        initialMoveOK.isGuardOK = (event: TouchEvent): boolean => {
+            // Interaction data has to be updated before testing the transition condition, since the condition relies on the data.
+            dataHandler?.onMove(event);
```

Comment:
I think this will not work: moveCondition checks the ID of the touch. If the ID is not the good one, the data are still updated (while it should not be).
Maybe check before this line and put the bool result in a var. If true call onMove. Return the bool.

Reply:
Sam thing for all the isGuardOK here